addons/web_editor/static/src/components/media_dialog/media_dialog.js

```diff
@@ -185,6 +185,12 @@ export class MediaDialog extends Component {
                             element.dataset.shapeColors = this.props.media.dataset.shapeColors;
                         }
                     }
+                    else if ([TABS.VIDEOS.id, TABS.DOCUMENTS.id].includes(this.state.activeTab)) {
+                        const parentEl = this.props.media.parentElement;
+                        if (parentEl.tagName === "A") {
```

Comment:
It's not enough; also check that the image is the only child of the link. Without it, for instance, if you replace a document that is inside a link (see image), the link and the text of the link will be removed.
![image](https://github.com/user-attachments/assets/f2d009ba-c85c-484d-a841-f9dee524c369)
I would also add a check to verify that `this.props.media` is an image.